test/functional/Java8andUp/src/org/openj9/test/utf8ClassPkgName/LoadClassWithUTF8PkgNameTest.java

```diff
@@ -0,0 +1,146 @@
+	final static String loadedClassNameWithDotPath = loadedPackagePath.replace('/', '.') + loadedClassName;
+	final static String jarFileNameWithPath = System.getProperty("user.dir") + "/" + "Utf8PkgName.jar";
+	
+	public static byte[] generateClassBytes(String loadedPackagePath, String loadedClassName, int classVersion) throws Exception {
+
+		ClassWriter cw = new ClassWriter(0);
+		FieldVisitor fv;
+		MethodVisitor mv;
+		AnnotationVisitor av0;
+
+		cw.visit(classVersion, ACC_PUBLIC + ACC_SUPER, loadedPackagePath + loadedClassName, null, "java/lang/Object", null);
+		cw.visitSource(loadedClassName + ".java", null);
+
+		{
+			mv = cw.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
+			mv.visitCode();
+			Label l0 = new Label();
+			mv.visitLabel(l0);
+			mv.visitLineNumber(29, l0);
+			mv.visitVarInsn(ALOAD, 0);
+			mv.visitMethodInsn(INVOKESPECIAL, "java/lang/Object", "<init>", "()V");
+			mv.visitInsn(RETURN);
+			Label l1 = new Label();
+			mv.visitLabel(l1);
+			mv.visitLocalVariable("this", "L" + loadedPackagePath + loadedClassName + ";", null, l0, l1, 0);
+			mv.visitMaxs(1, 1);
+			mv.visitEnd();
+		}
+		cw.visitEnd();
+
+		return cw.toByteArray();
+	}
+	
+	@BeforeTest
+	public void beforeTest() throws Exception {
+		String ver = System.getProperty("java.version");
+		boolean isJava8ClassFile = ((ver.length()>=3) && ver.substring(0, 2).equalsIgnoreCase("1.8")) ? true : false;
+		
+		/* The generated class file is only used by the java version >= Java 8 */
+		int classVersion = classVersionJava8;
+		if (!isJava8ClassFile) {
+			/* use the first located number in the version string assuming
+			 * the major version number is 9, 10, 11, 12, 13, 14 ... 
+			 * Note: internal build might come with '-' in the version string.
+			 * e.g. 14-internal
+			 */
+			String splitPattern = (ver.indexOf('.') >= 0) ? Pattern.quote(".") : 
+									((ver.indexOf('-') >=0) ? Pattern.quote("-"): null);
+			String verDigits = (null != splitPattern) ? ver.split(splitPattern)[0] : ver;
+			classVersion = (Integer.parseInt(verDigits) - 8) + classVersionJava8;
+		}
+		classBytes = generateClassBytes(loadedPackagePath, loadedClassName, classVersion);
+		
+		/* prepare the MANIFEST data for the specified jar file */
+		String jarFileVersion = "1.0.0";
+		String jarFileAuthor = "IBM OpenJ9";
+		Manifest manifestData = new Manifest();
+		Attributes manifestAttrs = manifestData.getMainAttributes();
+		manifestAttrs.put(Attributes.Name.MANIFEST_VERSION, jarFileVersion);
+		manifestAttrs.put(new Attributes.Name("Created-By"), jarFileAuthor);
+		
+		/* write the class bytes into the specified jar file */
+		OutputStream fos = new FileOutputStream(new File(jarFileNameWithPath));
+		JarOutputStream jos = new JarOutputStream(fos, manifestData);
+		JarEntry jarEntry = new JarEntry(loadedClassFileWithPackagePath);
+		jos.putNextEntry(jarEntry);
+		jos.write(classBytes, 0, classBytes.length);
+		jos.closeEntry();
+		jos.close();
+	}
```

Comment:
Pls use File.deleteOnExit() to remove the jar file after the test is finished.

Reply:
Not `try {} finally` should be used to ensure deleteOnExit() is called even if something goes wrong.

Reply:
Or call deleteOnExit() before writing to the file.

Reply:
The jar file is generated temporarily inside `.../openj9/test/TKG/test_output_xxxx` (xxxx means random digits) which should be cleaned up by the test framework.

Reply:
ok sounds good.